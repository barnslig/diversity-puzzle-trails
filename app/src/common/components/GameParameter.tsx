import { ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import * as React from "react";

type GameParameterProps = {
  /**
   * A representing icon, usually from @material-ui/icons
   */
  icon: React.ReactNode;

  /**
   * Current parameter value, e.g. 7
   */
  value: React.ReactNode;

  /**
   * Optional parameter label, e.g. Charakterpunkte
   */
  label?: React.ReactNode;
};

/**
 * A single game parameter
 */
const GameParameter = ({ icon, value, label }: GameParameterProps) => (
  <ListItem>
    <ListItemIcon>{icon}</ListItemIcon>
    <ListItemText primary={value} secondary={label} />
  </ListItem>
);

export default GameParameter;
