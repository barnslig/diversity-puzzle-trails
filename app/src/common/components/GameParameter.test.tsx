import * as React from "react";
import render from "../testing/render";

import GameParameter from "./GameParameter";

it("renders correctly", () => {
  const { container } = render(
    <GameParameter icon="" value="37" label="Bewegungspunkte" />
  );
  expect(container.firstChild).toMatchSnapshot();
});
