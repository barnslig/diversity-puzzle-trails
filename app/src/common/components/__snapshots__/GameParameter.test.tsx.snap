// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
>
  <div
    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  />
  <div
    class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      37
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
    >
      Bewegungspunkte
    </p>
  </div>
</li>
`;
